src/common/IComponent.hpp

```diff
@@ -105,6 +105,8 @@ class IComponent {
    */
   virtual ~IMeta() = default;
 
+  IMeta() = default;
```

Comment:
This seems useless, and even if you want a IMeta constructor it shall be protected and documented.